server/app/mutations/stacks/create.rb

```diff
@@ -45,11 +51,28 @@ def validate_services
       add_error("services.#{error.service}.links", :recursive, error.message)
     end
 
+    def validate_parent
+      _, parent_name = if self.inputs.has_key?(:parent)
+        self.inputs.delete(:parent)[:id].split('/')
+      else
+        [self.grid.name, self.inputs.delete(:parent_name)]
+      end
+      return unless parent_name
+
+      parent_stack = self.grid.stacks.find_by(name: parent_name)
+      if parent_stack.nil?
+        add_error(:parent, :not_found, "parent stack #{parent_name} does not exist")
```

Comment:
No. This fundamentally breaks the parent-child dependent stack relationship, because as @kke said, the child stacks are installed before the parent stacks.

```
> Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo ..
Value for peers is not set. Substituting with an empty string.
 [fail] Creating stack deptest-foo      
 [error] 422 : Unprocessable Entity:
	parent: parent stack deptest does not exist
```

Reply:
Easy to fix.

Reply:
I'm not sure what you mean.

See the original discussion around this exact same thing: https://github.com/kontena/kontena/pull/2707#discussion_r138290028

Reply:
I meant reverting this change. Currently it's not possible to validate parent because it's created afterwards.